### PR TITLE
Updated global section

### PR DESCRIPTION
**Pull request by cabidop**
This is a revamp of the global section component flow that simplifies it and includes explicit error handling when manually updating the JSON fields. It is also in preparation to a new feature to be included soon in the section (menu handling).

This also replaces both textarea inputs by a custom component based on the `AceEditor` that uses the JSON mode by default (which is not included in the base Discourse installation).

### CODE REVIEW

File: assets/javascripts/discourse/components/json-editor.js.es6 [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import AceEditor from "admin/components/ace-editor";
+
+export default class JsonEditor extends AceEditor {
+  mode = "json";
+  htmlPlaceholder = true;
+
+  _overridePlaceholder(loadedAce) {
```
Comment (cabidop) [inline assets/javascripts/discourse/components/json-editor.js.es6:7]: @angusmcleod this is the simplest way I found to plug into the Ace editor configuration from the Discourse component (without fully copying it), though it feels quite "hacky" as you can see.

Although it is nice to have the JSON syntax highlighting, it is also an option to just use the plain version of the editor instead and not include the additional Ace related files.

Comment (angusmcleod) [reply assets/javascripts/discourse/components/json-editor.js.es6:7]: I'm happy with this extension of the core class. Slightly hacky, but not that bad in the scheme of things. You have to do this kind of thing sometimes in a plugin.

**Comment by angusmcleod** [comment]
@cabidop I like this! The only thing I'll ask here is that we use this opportunity to add a qunit acceptance test for the new editor. As mentioned in my comment, I think extending the core editor in that way is fine. However I think perhaps we should also add at least one acceptance test for it. And we need to start adding acceptance tests somewhere with this plugin.

**Review by angusmcleod — CHANGES_REQUESTED** [review]
@cabidop I like this! The only thing I'll ask here is that we use this opportunity to add a qunit acceptance test for the new editor. As mentioned in my comment, I think extending the core editor in that way is fine. However I think perhaps we should also add at least one acceptance test for it. And we need to start adding frontend tests somewhere with this plugin.

File: test/javascripts/acceptance/global-json-editor-test.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { test } from "qunit";
+import {
+  click,
+  fillIn,
+  triggerEvent,
+  visit,
+  waitFor,
+} from "@ember/test-helpers";
+
+acceptance("Global | JSON editor", function (needs) {
```
[the remark this answers is not in the document]
Comment (angusmcleod) [reply test/javascripts/acceptance/global-json-editor-test.js:15]: @cabidop Looks good! Nice work.

**Review by angusmcleod — APPROVED** [review]

**Pull request merged**